src/plugins/index_pattern_management/server/plugin.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { PluginInitializerContext, CoreSetup, Plugin } from 'src/core/server';
+import { schema } from '@kbn/config-schema';
+
+export class IndexPatternManagementPlugin implements Plugin<void, void> {
+  constructor(initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+
+    router.get(
+      {
+        path: `/api/index-pattern-management/resolve_index/{query}`,
```

Comment:
I think we should consider making this route `/internal` instead of `/api`.

If we aren't expecting other folks to use this, we should definitely make it an internal route so there's no confusion around stability guarantees. 

And if we _are_ expecting other folks to need to use this, perhaps we should still be using an internal REST API, and then providing a lightweight client for the browser that can be used instead.